### ISSUE TITLE
[Problem]: Unknown PCM default

### ISSUE THREAD

**Issue by cyberb** [mikebrady/shairport-sync#1686]
### What happened?

I am trying to run shairport for Syncloud project (https://github.com/syncloud/platform) where I am packaging apps in a snap format which means all dependencies have to come with the package and nothing is installed to the system.

I have running nqptp (under root) and shairport under shairport user, but as soon as I try to test it with an android app (AirMusic Trial) shairport restarts with this message:

I do not have alsa installed in the system and I am using the docker image (https://github.com/mikebrady/shairport-sync-docker) to take all the binaries/libraries to run using the linker directly.

Syncloud app package: https://github.com/syncloud/shairport

Thanks

### Relevant log output

```shell
Jun 08 17:59:52 syncloud shairport.shairport[26139]: ALSA lib conf.c:4555:(snd_config_update_r) Cannot access file /usr/share/alsa/alsa.conf
Jun 08 17:59:52 syncloud shairport.shairport[26139]: ALSA lib pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM default
Jun 08 17:59:52 syncloud shairport.shairport[26139]: fatal error: An unrecoverable error, "output_device_error_2", has been detected. Doing an emergency exit, as no run_t
```

### Configuration Information.

>> Display Config Start.

From "uname -a":
 Linux syncloud 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1 (2020-01-26) x86_64 GNU/Linux

From /etc/os-release:
 Debian GNU/Linux 10 (buster)

Shairport Sync Version String:
 4.2-dirty-AirPlay2-smi9-alac-OpenSSL-Avahi-ALSA-pa-dummy-stdout-pipe-soxr-convolution-metadata-mqtt-dbus-mpris-sysconfdir:/etc

Command Line:
 /snap/shairport/25/shairport/usr/local/bin/shairport-sync --displayConfig
No configuration file.

>> Display Config End.
>> Goodbye!

### How did you install Shairport Sync?

Built from source

### Check previous issues

- [X] Confirm

**Comment by cyberb** [comment]
after linking alsa conf dir I got further
```
ln -s /snap/shairport/current/shairport/usr/share/alsa /usr/share/alsa
```
new error:
```
Jun 08 18:36:38 syncloud shairport.shairport[29656]: ALSA lib confmisc.c:855:(parse_card) cannot find card '0'
Jun 08 18:36:38 syncloud shairport.shairport[29656]: ALSA lib conf.c:5180:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory
Jun 08 18:36:38 syncloud shairport.shairport[29656]: ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings
Jun 08 18:36:38 syncloud shairport.shairport[29656]: ALSA lib conf.c:5180:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
Jun 08 18:36:38 syncloud shairport.shairport[29656]: ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name
Jun 08 18:36:38 syncloud shairport.shairport[29656]: ALSA lib conf.c:5180:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
Jun 08 18:36:38 syncloud shairport.shairport[29656]: ALSA lib conf.c:5703:(snd_config_expand) Evaluate error: No such file or directory
Jun 08 18:36:38 syncloud shairport.shairport[29656]: ALSA lib pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM default
Jun 08 18:36:38 syncloud shairport.shairport[29656]: fatal error: An unrecoverable error, "output_device_error_2", has been detected. Doing an emergency exit, as no run_t

```

**Comment by mikebrady** [comment]
Thanks for the post. Shairport Sync needs an audio output device and that is usually accessed via the `alsa` subsystem. The Docker image does not contain the `alsa` subsystem -- the host's `alsa` subsystem is used instead. So you'd have to something similar to get Shairport Sync to work in your system. Unfortunately, I am not sufficiently expert in `alsa` to suggest how that should be done.

**Comment by cyberb** [comment]
OMG it works!
I added shairport user to audio and raspberrypi instantly started to work.
Something I learned today, Alsa is a server-less lib so no need to install anything and libs from docker image are enough, ideally there is a way to not have its configs in /usr/share/alsa, but I will have to ask this question somewhere else. I guess that is the reason to have PulseAudio which is a server in front of alsa lib.

Last question, ideally for self hosting (of multiple apps) I prefer unix sockets instead of tcp ports (NQPTP), is it possible to have it?
